databases/tasks_store.go

```diff
@@ -0,0 +1,298 @@
+	}
+
+	return int(id), nil
+}
+
+// GetTasks returns a slice of models.TaskFromDB and an error.
+//
+// The function takes one string parameter for a search query.
+//
+// If the search query is a date in the format "DD.MM.YYYY", the function
+// returns a slice of tasks for that date. Otherwise, the function returns a
+// slice of tasks whose title or comment contains the search query.
+//
+// The returned slice of tasks is limited to 20 tasks.
+//
+// If the tasks cannot be retrieved, the function returns an error with the
+// following format: "can't get tasks: <error message>".
+func GetTasks(search string) ([]models.TaskFromDB, error) {
+	database, err := sql.Open("sqlite", path)
+	if err != nil {
+		return nil, fmt.Errorf("can't open database: %s", err.Error())
+	}
+	defer database.Close()
+
+	dateRegExp := regexp.MustCompile(`^([0-2][0-9]|(3)[0-1])\.(0[1-9]|1[0-2])\.\d{4}$`)
+	matched := dateRegExp.MatchString(search)
+
+	var tasks = make([]models.TaskFromDB, 0, 20)
+	var rows *sql.Rows
+
+	switch matched {
+	case true:
+		date, err := time.Parse("02.01.2006", search)
+		if err != nil {
+			return nil, fmt.Errorf("error while parsing date: %v", err)
+		}
+
+		dateStr := date.Format("20060102")
+
+		rows, err = database.Query(`SELECT id, date, title, comment, repeat FROM scheduler
+			WHERE date = :date ORDER BY date ASC LIMIT 20`,
+			sql.Named("date", dateStr))
+		if err != nil {
+			return nil, fmt.Errorf("can't get tasks: %v", err)
+		}
+	default:
+		rows, err = database.Query(`SELECT id, date, title, comment, repeat FROM scheduler
+			WHERE title LIKE :search OR comment LIKE :search ORDER BY date ASC LIMIT 20`,
+			sql.Named("search", "%"+search+"%"))
+		if err != nil {
+			return nil, fmt.Errorf("can't get tasks: %v", err)
+		}
+	}
+	defer rows.Close()
+
+	for rows.Next() {
+		var task models.TaskFromDB
+		err = rows.Scan(&task.ID, &task.Date, &task.Title, &task.Comment, &task.Repeat)
+		if err != nil {
+			return nil, fmt.Errorf("can't scan task: %s", err.Error())
+		}
+
+		tasks = append(tasks, task)
+	}
+
+	if err := rows.Err(); err != nil {
+		return nil, fmt.Errorf("error while reading rows: %s", err.Error())
+	}
+
+	return tasks, nil
+}
+
+// GetTaskByID gets a task from the database by its id.
+//
+// The function takes a string id parameter and returns a TaskFromDB struct and an error.
+// If the task is not found, the error is sql.ErrNoRows.
+// If there is a database error, the error is wrapped with a message in the format
+// "can't get task: <error message>".
+func GetTaskByID(id string) (models.TaskFromDB, error) {
+	var task models.TaskFromDB
+
+	database, err := sql.Open("sqlite", path)
+	if err != nil {
+		return task, fmt.Errorf("can't open database: %v", err)
+	}
+	defer database.Close()
+
+	err = database.QueryRow(`SELECT id, date, title, comment, repeat FROM scheduler WHERE id = :id`,
+		sql.Named("id", id)).Scan(&task.ID, &task.Date, &task.Title, &task.Comment, &task.Repeat)
+
+	if err == sql.ErrNoRows {
```

Comment:
Лучше сравнивать ошибки так. При оборачивании простое сравнение сломается
```suggestion
	if errors.Is(err, sql.ErrNoRows) {
```